db/mem_batch.go

```diff
@@ -35,16 +37,29 @@ func (mBatch *memBatch) Delete(key []byte) {
 }
 
 func (mBatch *memBatch) Write() {
+	mBatch.WriteSync()
+}
+
+func (mBatch *memBatch) WriteSync() {
 	mtx := mBatch.db.Mutex()
 	mtx.Lock()
 	defer mtx.Unlock()
 
-	for _, op := range mBatch.ops {
-		switch op.opType {
-		case opTypeSet:
-			mBatch.db.SetNoLock(op.key, op.value)
-		case opTypeDelete:
-			mBatch.db.DeleteNoLock(op.key)
+	for i, op := range mBatch.ops {
+		if i < len(mBatch.ops) {
```

Comment:
I don't understand how's i can be greater than `len(mBatch.ops)` if we're ranging over it

Reply:
oops, you're right, I missed a -1 . I'll add a test.